lib/gems/pending/util/object_storage/miq_s3_storage.rb

```diff
@@ -45,7 +45,14 @@ def download_single(source, destination)
 
     with_standard_s3_error_handling("downloading", source) do
       if destination.kind_of?(IO) || destination.kind_of?(StringIO)
-        s3.client.get_object(:bucket => bucket_name, :key => object_key) do |chunk|
+        get_object_opts = {
+          :bucket => bucket_name,
+          :key    => object_key
+        }
+        # :range is indexed starting at zero
+        get_object_opts[:range] = "bytes=0-#{byte_count - 1}" if byte_count
```

Comment:
Should this range always start at 0? How does that work after the first part is downloaded?

Reply:
@carbonin sorry, as it stands right now, this is only really being used for magic number fetching, as the link [in my comment](https://github.com/ManageIQ/manageiq-gems-pending/pull/399#issuecomment-435997220) to @jerryk55 was trying to show.

Basically it is for only getting the first couple of bits for a magic number for the "yet to be opened" PR for `MiqFileStorage#magic_number_for`.

Reply:
Okay, maybe just a comment then to track that this is really `download_first` rather than `download_single` when byte count is passed.